llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -5037,6 +5107,7 @@ Function *SPIRVToLLVM::transFunction(SPIRVFunction *bf) {
   FunctionType *ft = dyn_cast<FunctionType>(transType(bf->getFunctionType()));
   Function *f = dyn_cast<Function>(mapValue(bf, Function::Create(ft, linkage, bf->getName(), m_m)));
   assert(f);
+  continueAtLastInsertPoint = false;
```

Comment:
Thanks - that would avoid the issue with code being inserted in a wrong function.

Would it also work if you placed this line in [the beginning of the "for" loop ](https://github.com/GPUOpen-Drivers/llpc/pull/1260/commits/19b9d7d62001eab03132d33d5dffccd749cef5bb#diff-5d44c53248894af806173ad0baa60912de54085dc6ac296a86d8bf69f3fa8f44R5149 ) instead of placing it at the end of that loop and here? If so, that would be preferable (fewer accesses of a global var).